scripts/exgdas_efsoi.sh

```diff
@@ -0,0 +1,388 @@
+#ATMGES_ENSMEAN=$COMIN_GES_ENS/${GPREFIX}atmf006.ensmean${GSUFFIX}
+ATMGES_ENSMEAN=$COMIN_ANL/$VERFANL
+if [ $SUFFIX = ".nc" ]; then
+   LONB_ENKF=${LONB_ENKF:-$($NCLEN $ATMGES_ENSMEAN grid_xt)} # get LONB_ENKF
+   LATB_ENKF=${LATB_ENKF:-$($NCLEN $ATMGES_ENSMEAN grid_yt)} # get LATB_ENFK
+   LEVS_ENKF=${LEVS_ENKF:-$($NCLEN $ATMGES_ENSMEAN pfull)} # get LEVS_ENFK
+   use_gfs_ncio=".true."
+   use_gfs_nemsio=".false."
+   paranc=${paranc:-".true."}
+   if [ $DO_CALC_INCREMENT = "YES" ]; then
+      write_fv3_incr=".false."
+   else
+      write_fv3_incr=".true."
+      WRITE_INCR_ZERO="incvars_to_zero= $INCREMENTS_TO_ZERO,"
+   fi
+else
+   LEVS_ENKF=${LEVS_ENKF:-$($NEMSIOGET $ATMGES_ENSMEAN dimz | awk '{print $2}')}
+   LATB_ENKF=${LATB_ENKF:-$($NEMSIOGET $ATMGES_ENSMEAN dimy | awk '{print $2}')}
+   LONB_ENKF=${LONB_ENKF:-$($NEMSIOGET $ATMGES_ENSMEAN dimx | awk '{print $2}')}
+   use_gfs_ncio=".false."
+   use_gfs_nemsio=".true."
+   paranc=${paranc:-".false."}
+fi
+LATA_ENKF=${LATA_ENKF:-$LATB_ENKF}
+LONA_ENKF=${LONA_ENKF:-$LONB_ENKF}
+
+SATANGL=${SATANGL:-${FIXgsi}/global_satangbias.txt}
+SATINFO=${SATINFO:-${FIXgsi}/global_satinfo.txt}
+CONVINFO=${CONVINFO:-${FIXgsi}/global_convinfo.txt}
+OZINFO=${OZINFO:-${FIXgsi}/global_ozinfo.txt}
+SCANINFO=${SCANINFO:-${FIXgsi}/global_scaninfo.txt}
+HYBENSINFO=${HYBENSINFO:-${FIXgsi}/global_hybens_info.l${LEVS_ENKF}.txt}
+ANAVINFO=${ANAVINFO:-${FIXgsi}/global_anavinfo.l${LEVS_ENKF}.txt}
+VLOCALEIG=${VLOCALEIG:-${FIXgsi}/vlocal_eig_l${LEVS_ENKF}.dat}
+
+ENKF_SUFFIX="s"
+[[ $SMOOTH_ENKF = "NO" ]] && ENKF_SUFFIX=""
+
+################################################################################
+# Preprocessing
+mkdata=NO
+if [ ! -d $DATA ]; then
+   mkdata=YES
+   mkdir -p $DATA
+fi
+cd $DATA || exit 99
+
+################################################################################
+# Fixed files
+$NLN $SATANGL    satbias_angle
+$NLN $SATINFO    satinfo
+$NLN $SCANINFO   scaninfo
+$NLN $CONVINFO   convinfo
+$NLN $OZINFO     ozinfo
+$NLN $HYBENSINFO hybens_info
+$NLN $ANAVINFO   anavinfo
+$NLN $VLOCALEIG  vlocal_eig.dat
+
+# Bias correction coefficients based on the ensemble mean
+$NLN $COMOUT_ANL_ENSFSOI/$GBIASe satbias_in
+
+################################################################################
+# Ensemble guess, observational data and analyses/increments
+
+nfhrs=`echo $IAUFHRS_ENKF | sed 's/,/ /g'`
+for imem in $(seq 1 $NMEM_ENKF); do
+   memchar="mem"$(printf %03i $imem)
+   mkdir ${memchar}
+   $NLN $COMOUT_ANL_ENSFSOI/$memchar/${APREFIX}atmf024.nc ${memchar}
+done
+
+$NLN $COMIN_GES_ENS/${GPREFIX}atmf006.ensmean${GSUFFIX} sfg_${CDATE}_fhr06_ensmean
+$NLN $COMIN_GES_ENS/${GPREFIX}atmf006.ensmean${GSUFFIX} sfg_${CDATE}_fhr03_ensmean
+
+# verifying analysis
+$NLN ${COMIN_ANL}/${VERFANL} .
+# the above way is the proper one, but this bit of subterfuge
+# is necessary because ensmean is hard coded into GSI - needs
+# to changed
+$NLN ${COMIN_ANL}/${VERFANL} ${APREFIX}atmanl.ensmean.nc
```

Comment:
@AndrewEichmann-NOAA Can you explain in more detail what's happening here?  Where in the GSI are you referring?

Reply:
@CatherineThomas-NOAA @MichaelLueken-NOAA As discussed it regards localization advection: currently it takes the mean of the advection field between the f024 and the verifying analysis, but should be ideally the ensemble mean initializing analysis. Because of the changes that would require in code more important than that, I am working on using the GDAS ensres analysis, but because this is not critical it can wait.